hamlet/command/release/__init__.py

```diff
@@ -0,0 +1,152 @@
+import click
+import functools
+
+from hamlet.command import root as cli
+from hamlet.command.common import exceptions, config
+
+from hamlet.backend.automation_tasks import upload_image as upload_image_backend
+from hamlet.backend.automation_tasks import transfer_image as transfer_image_backend
+from hamlet.backend.automation_tasks import update_build as update_build_backend
+
+
+def image_options(func):
+    """Standard image details"""
+
+    @click.option(
+        "-u",
+        "--deployment-unit",
+        envvar="DEPLOYMENT_UNIT",
+        required=True,
+        help="The deployment unit the image belongs to",
+    )
+    @click.option(
+        "-r",
+        "--build-reference",
+        envvar="BUILD_REFERENCE",
+        required=True,
+        help="The unique reference for the build of this image - usually git commit",
+    )
+    @click.option(
+        "--code-tag",
+        envvar="CODE_TAG",
+        help="A tag applied to your code repo which will be mapped to a build reference",
+    )
+    @click.option(
+        "-f",
+        "--image-format",
+        envvar="IMAGE_FORMAT",
+        required=True,
```

Comment:
I think maybe we start with this as optional and let the script behind it flag an error if no format is provided and it can't work out which one to use.

For the stream jobs, they just deal with units and build references, and rely on the shared build configuration to provide scope and format information. 

It keeps the build.json files limited to just the reference, with the other information ready to be migrated to registry lookup within the cmdb.